scipy/stats/_bootstrap.py

```diff
@@ -0,0 +1,393 @@
+    distributions approximately `confidence_level`:math:`\, \times \, n` times.
+
+    Parameters
+    ----------
+    data : sequence of array-like
+         Each element of data is a sample from an underlying distribution.
+    statistic : callable
+        Statistic for which the confidence interval is to be calculated.
+        `statistic` must be a callable that accepts ``len(data)`` samples
+        as separate arguments and returns the resulting statistic.
+        `statistic` must also accept a keyword argument `axis` and be
+        vectorized to compute the statistic along the provided `axis`.
+    axis : int, optional
+        The axis of the samples in `data` along which the `statistic` is
+        calculated. The default is ``0``.
+    confidence_level : float, optional
+        The confidence level of the confidence interval.
+        The default is ``0.95``.
+    n_resamples : int, optional
+        The number of resamples performed to form the bootstrap distribution
+        of the statistic. The default is ``9999``.
+    method : str in {'percentile', 'basic', 'bca'}, optional
+        Whether to return the 'percentile' bootstrap confidence interval
+        (``'percentile'``), the 'reverse'or the bias-corrected and accelerated
+        bootstrap confidence interval (``'BCa'``). The default is ``'BCa'``.
+        Note that only ``'percentile'`` and ``'basic'`` support multi-sample
+        statistics at this time.
+    random_state : {None, int, `numpy.random.Generator`,
+                    `numpy.random.RandomState`}, optional
+
+        If `seed` is None (or `np.random`), the `numpy.random.RandomState`
+        singleton is used.
+        If `seed` is an int, a new ``RandomState`` instance is used,
+        seeded with `seed`.
+        If `seed` is already a ``Generator`` or ``RandomState`` instance then
+        that instance is used.
+
+        Pseudorandom number generator state used to generate resamples.
+
+    Returns
+    -------
+    res : BootstrapResult
+        An object with attributes:
+
+        confidence_interval : ConfidenceInterval
+            The bootstrap confidence interval as an instance of
+            `collections.namedtuple` with attributes `low` and `high`.
+        standard_error : float or ndarray
+            The bootstrap standard error, that is, the sample standard
+            deviation of the bootstrap distribution
+
+    References
+    ----------
+    .. [1] B. Efron and R. J. Tibshirani, An Introduction to the Bootstrap,
+       Chapman & Hall/CRC, Boca Raton, FL, USA (1993)
+    .. [2] Nathaniel E. Helwig, "Bootstrap Confidence Intervals",
+       http://users.stat.umn.edu/~helwig/notes/bootci-Notes.pdf
+    .. [3] Bootstrapping (statistics), Wikipedia,
+       https://en.wikipedia.org/wiki/Bootstrapping_(statistics)
+
+    Examples
+    --------
+    Suppose we have sampled data from an unknown distribution.
+
+    >>> import numpy as np
+    >>> rng = np.random.default_rng()
```

Comment:
@sturlamolden please see gh-13863.

Reply:
Gh-13863 and predecessors are wrong. Though the default Generator in NumPy 1.17 is PCG64, the documentation says this can change in future versions. And if this happens, the doctests and test suite breaks.

Reply:
Added a comment to gh-13778

Reply:
OK. Let's consider the issue in this PR resolved, though. I've already changed it from an explicit random state to conform, so if we need to change things back, all that can be done at the same time (presumably after this merges).

Reply:
I replied as well to your comment on the issue @sturlamolden 
https://github.com/scipy/scipy/pull/13778#issuecomment-837300651